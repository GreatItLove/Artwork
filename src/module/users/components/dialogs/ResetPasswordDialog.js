import React from 'react';
import Dialog from 'material-ui/Dialog';
import PropTypes from 'prop-types';

import ResetPassword from '../../containers/ResetPassword';

const propTypes = {
  resettingPassword: PropTypes.bool.isRequired,
  userId: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]).isRequired,
  closeResetPasswordDialog: PropTypes.func.isRequired
};

/**
 * @function ResetPasswordDialog
 * @returns {XML}
 */
function ResetPasswordDialog({ resettingPassword, userId, closeResetPasswordDialog }) {
  return (
    <Dialog
      onRequestClose={closeResetPasswordDialog}
      open={resettingPassword}
      bodyStyle={{ overflowY: 'unset' }}
      overlayStyle={{ backgroundColor: '#797979', opacity: '0.8' }}
      paperProps={{ style: { backgroundColor: 'none' }, zDepth: 0 }}
    >
      {resettingPassword && <ResetPassword userId={userId} closeDialog={closeResetPasswordDialog} />}
    </Dialog>
  );
}

ResetPasswordDialog.propTypes = propTypes;

export default ResetPasswordDialog;
