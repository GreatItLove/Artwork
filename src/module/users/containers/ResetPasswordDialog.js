import { connect } from 'react-redux';

import { closeUserResetPasswordDialog } from '../usersActions';
import ResetPasswordDialog from '../components/dialogs/ResetPasswordDialog';

const mapStateToProps = state => ({
  resettingPassword: state.users.resettingPassword !== false,
  userId: state.users.resettingPassword
});

const mapDispatchToProps = dispatch => ({
  closeResetPasswordDialog: () => {
    dispatch(closeUserResetPasswordDialog());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(ResetPasswordDialog);
