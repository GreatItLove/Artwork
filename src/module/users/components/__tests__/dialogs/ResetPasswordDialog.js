import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ResetPasswordDialog from '../../dialogs/ResetPasswordDialog';

describe('ResetPasswordDialog', () => {
  it('Snapshot of ResetPasswordDialog', () => {
    const props = {
      resettingPassword: false,
      userId: 546546,
      closeResetPasswordDialog: jest.fn()
    };
    const wrapper = shallow(<ResetPasswordDialog {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
