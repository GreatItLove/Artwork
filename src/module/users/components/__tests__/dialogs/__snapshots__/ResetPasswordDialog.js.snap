// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordDialog Snapshot of ResetPasswordDialog 1`] = `
<Dialog
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  bodyStyle={
    Object {
      "overflowY": "unset",
    }
  }
  modal={false}
  onRequestClose={[Function]}
  open={false}
  overlayStyle={
    Object {
      "backgroundColor": "#797979",
      "opacity": "0.8",
    }
  }
  paperProps={
    Object {
      "style": Object {
        "backgroundColor": "none",
      },
      "zDepth": 0,
    }
  }
  repositionOnUpdate={true}
/>
`;
